test/images/pets/mysql/main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type server struct {
+	mux            *http.ServeMux
+	readDBAddress  string
+	writeDBAddress string
+}
+
+func newServer(readDBAddress, writeDBAddress string) (*server, error) {
+	s := &server{
+		mux:            http.NewServeMux(),
+		readDBAddress:  readDBAddress,
+		writeDBAddress: writeDBAddress,
+	}
+
+	s.mux.HandleFunc("/addName", func(w http.ResponseWriter, r *http.Request) {
+		name := r.FormValue("name")
+		if name == "" {
+			http.Error(w, "name must be specified", http.StatusBadRequest)
+			return
+		}
+		if err := s.writeName(name); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+	})
+
+	s.mux.HandleFunc("/countNames", func(w http.ResponseWriter, r *http.Request) {
+		numNames, err := s.countNames()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		json.NewEncoder(w).Encode(numNames)
+	})
+
+	s.mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	return s, nil
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.mux.ServeHTTP(w, r)
+}
+
+func (s *server) createDBConnection(read bool) (*sql.DB, error) {
+	addr := s.writeDBAddress
+	if read {
+		addr = s.readDBAddress
+	}
+	db, err := sql.Open("mysql", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := db.Ping(); err != nil {
+		return nil, err
+	}
+	return db, nil
+
+}
+
+func (s *server) writeName(name string) error {
+	db, err := s.createDBConnection(false)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+	_, err = db.Exec("INSERT INTO testing.users (name) VALUEs(?);", name)
+	return err
+}
+
+func (s *server) countNames() (int, error) {
+	db, err := s.createDBConnection(true)
+	if err != nil {
+		return 0, err
+	}
+	defer db.Close()
+
+	rows, err := db.Query("SELECT * FROM testing.users;")
```

Comment:
Minor nit, but why not SELECT COUNT(*) and trust the database?  Maybe because we're trying to simulate more throughput?